### ISSUE TITLE
The Bluetooth device icon does not become active or inactive immediately

### ISSUE THREAD

**Issue by ipwin-github** [SARDONYX-sard/bluetooth-battery-monitor#15]
The Bluetooth device icon in the main window of the program becomes active or inactive only after the set update interval has expired. 
For example, if the default update interval is 60 minutes, the device icon will become active only after 60 minutes.  Before that, the device icon will be crossed out, as if the device is not connected. When you turn off your Bluetooth device, the program also does not update the icon immediately.
It would be nice if the program could immediately detect when the Bluetooth device is on or off.

**Comment by SARDONYX-sard** [comment]
Yes, we are currently using [Script](https://github.com/SARDONYX-sard/bluetooth-battery-monitor/blob/0.3.1/crates/bluetooth/scripts/get-bluetooth-battery-all.ps1) to get information on all devices (connection status, battery information, last used, etc.) in a constant interval sense and not in real time.

I am also looking for a way to do it in real time, but so far I am not sure.

https://github.com/SARDONYX-sard/bluetooth-battery-monitor/issues/11

**Comment by sergeivich** [comment]
maybe it's possible to have the interval auto changed upon device connection to something like 1-second, then after execution of task, the interval is auto changed back to previous setting.

**Comment by SARDONYX-sard** [comment]
I do not know how to detect the connection of a device in the first place...

I have tried the [DeviceWatcher](https://learn.microsoft.com/en-us/uwp/api/windows.devices.enumeration.devicewatcher?view=winrt-26100), but it has never worked.

**Comment by SARDONYX-sard** [comment]
I found a way to make DeviceWatcher work.
I think it will be possible to get update events automatically from the OS side instead of in interval format.

Will probably be available in v4. (I think you can get it from GitHub Action if you want to try it fast.
